mcstatus/status_response.py

```diff
@@ -0,0 +1,326 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Union
+
+STYLE_MAP = {
+    "bold": "l",
+    "italic": "o",
+    "underlined": "n",
+    "obfuscated": "k",
+    "color": {
+        "dark_red": "4",
+        "red": "c",
+        "gold": "6",
+        "yellow": "e",
+        "dark_green": "2",
+        "green": "a",
+        "aqua": "b",
+        "dark_aqua": "3",
+        "dark_blue": "1",
+        "blue": "9",
+        "light_purple": "d",
+        "dark_purple": "5",
+        "white": "f",
+        "gray": "7",
+        "dark_gray": "8",
+        "black": "0",
+    },
+}
+
+
+@dataclass
+class AbstractDataclass(ABC):
+    """Abstract base class for all dataclasses.
+
+    We can't use just `ABC` because we need raise TypeError if we try to
+    instantiate an abstract dataclass. Class which inherit from `ABC` raises
+    it only if it has abstract methods.
+    """
+
+    def __new__(cls, *args, **kwargs):
+        if cls == AbstractDataclass or cls.__bases__[0] == AbstractDataclass:
+            raise TypeError("Cannot instantiate abstract class.")
+        return super().__new__(cls)
+
+
+@dataclass
+class MCServerResponse(AbstractDataclass):
+    """Dataclass for storing cross-platform status answer object."""
+
+    players: ServerPlayers
+    version: ServerVersion
+    # Message Of The Day. Also known as `Description`.
```

Comment:
It true, until you will try transform dataclass to docs (example with sphinx).

If you specify description in docstring, this will just double documented item.

As example, how this looks like in sphinx:
```
class Example:
    """:attribute some: Description."""
    some: str
```
It will render to:
```md
## Atributes

- `some: str`
- `some`
Description
```
Example illustrative, and actually looks another. I just haven't now possible create mini-project and render docs for it.

Reply:
Yeah, we're currently using `:param [name]: description`, though there's no standardized way and there are other styles being used too, for now let's just move it to the docstring, no matter the style, that can be done with documentation once we settle on some concrete docstring standard.

Reply:
Fixed in fa7beed82688eab84709c9928a69ff7bffe58431.